glotaran/model/dataset.py

```diff
@@ -49,96 +58,198 @@ def get_calculated_axis(self):
         """Get the axis along the estimated matrices are grouped"""
         raise NotImplementedError
 
-    def get_axis(self, label):
+    def get_axis(self, axis_label: str) -> Union[list, np.ndarray]:
```

Comment:
the axis MUST be ndarrays

Reply:
another idea would be to make `get_axis` and `set_axis` privat ( `_get_axis` / `_set_axis` ) and let the user only see the custom implementation, as is done in `SpectralTemporalDataset` with `time_axis` and `spectral_axis`.

Reply:
I am generally not caring so much. I have set it now to ndarray, since I use excplicitly things like `axe.size`. The thing is, that an axe in our context must have a size which can be retrieved by a standard method somehow and it must contain hashable items (actually only for the calculated axis ('time' in terms of a kinetic model).

Leaving that aside, `Dataset` is an abstract class without ABC. It provides a standart implementation for convinience (escpecially simulations). I don't want to strictly split it into interface and implementation, but maybe I should. In that case, I would leave the setter for axies out and only put the getter into. I experimentated with abstract properties, that would be a nice way to go.

As for this PR. consider Dataset as an interface like that:

```python
class Dataset:
    get_axis(label: str) -> ndarray
    data() -> ndarray
```